liesel/distributions/mvn_degen.py

```diff
@@ -95,8 +103,37 @@ class MultivariateNormalDegenerate(tfd.Distribution):
       operation and can be avoided by specifying the corresponding arguments.
     * When you draw samples from the distribution via :meth:`.sample`, it is always
       necessary to compute the eigendecomposition of the distribution's precision
-      matrices once and cache it, because sampling requires both the eigenvalues
-      and eigenvectors.
+      matrices once and cache it, because sampling requires both the eigenvalues and
+      eigenvectors.
+
+    **Details on sampling**
+
+    If we have a singular precision matrix :math:`P`, we can view it as the
+    pseudo-inverse :math:`\\Sigma^+` of the variance-covariance matrix :math:`\\Sigma`.
+    We can obtain :math:`\\Sigma` by finding the singular value decomposition
+
+    .. math::
+        \\Sigma^+ = QA^+Q^T,
+
+    where :math:`Q` is the orthogonal matrix of eigenvectors and :math:`A^+` is the
+    diagonal matrix of singular values of :math:`\\Sigma^+ = P`. If the precision matrix
+    is singular, then :math:`\\text{diag}(A^+)` contains zeroes. Now we take the inverse
+    of the non-zero entries of :math:`\\text{diag}(A^+)`, while the zero entries remain
+    at zero, resulting in a matrix :math:`A`. We can now write
+
+    .. math::
+        \\Sigma = Q A Q^T.
+
+    To sample from :math:`N(\\mu, \\Sigma)`, we first draw a sample
+    :math:`z \\sim N(0, I)` from a standard normal distribution. Next, we transform the
+    sample by applying :math:`x = Q A^{1/2}z`, such that
+    :math:`\\text{Cov}(x) = \\Sigma`:
```

Comment:
Can't you remove the entries corresponding to zero EVs from Q and A and reduce the dimension of z?

Reply:
Yep, that leads to identical results. We could add that as an issue to further improve the efficiency of the sampling function. For the moment, I think the gain in efficiency would not be so big, so personally, I would prefer to leave the function as it is at the moment - it might be a bit of a haggle with the batch sizes again.